11-Refactor-the-simulator-to-be-multi-threaded pull request

Nadim-Muhammad opened the pull request:
Created a multi-threaded underlying price simulator and fixed a CRUCIAL BUG in simulator.cpp (and theorethical values..) which I did not saw earlier, which gave us wrong outputs. Im 99% confidend it should be correct now. @Staneesh Please check and merge if it's good.

Staneesh commented on src/simulator.cpp:
```diff
@@ -19,14 +19,55 @@ void StockSimulator::runSimulation() {
+
+    // Wait for all threads to finish
+    for (auto& thread : threads) {
+        thread.join();
```
Does `thread.join()` return some value that could be checked for determining if `join()` succeeded?

Staneesh commented on src/simulator.h:
```diff
@@ -21,6 +22,8 @@ class StockSimulator {
 
     void runSimulation();
 
+    void runSimulationParallel();
```
Please have a prompt in `main` asking the user if he wants to run tests or the single-threaded simulation or the multi-threaded simulation. Please also add a prompt for the maximum number of threads that the user wants to use. In the limit, the time spent running single-threaded simulation should equal 1-thread variant of the multi-threaded simulation. 

Limiting number of threads used is super useful if you don't want to overload your machine and keep doing other tasks.

Staneesh reviewed (changes requested):
Hey, left some notes below 🌴

Nadim-Muhammad commented:
Updated the 11-Refactor-the-simulator-to-be-multi-threaded branch, commented the code and did things you mentioned. Merged runSimulation() and runSimulationParallel() into one and added propmt to choose how many cores does user want to use. Check code again and merge if it's good.
